pkg/virt-handler/cache/cache.go

```diff
@@ -348,6 +360,41 @@ func (d *DomainWatcher) handleStaleWatchdogFiles() error {
 	return nil
 }
 
+func (d *DomainWatcher) handleResync() {
+
+	socketFiles, err := listSockets()
+	if err != nil {
+		log.Log.Reason(err).Error("failed to list sockets")
+		return
+	}
+
+	log.Log.Infof("resyncing virt-launcher domains")
+	for _, socket := range socketFiles {
+		client, err := cmdclient.NewClient(socket)
+		if err != nil {
+			log.Log.Reason(err).Error("failed to connect to cmd client socket during resync")
+			// Ignore failure to connect to client.
+			// These are all local connections via unix socket.
+			// A failure to connect means there's nothing on the other
+			// end listening.
+			continue
+		}
+		defer client.Close()
+
+		domain, exists, err := client.GetDomain()
+		if err != nil {
+			// this resync is best effort only.
+			log.Log.Reason(err).Errorf("unable to retrieve domain at socket %s during resync", socket)
+			continue
+		} else if !exists {
+			// nothing to sync if it doesn't exist
+			continue
```

Comment:
We still get the final delete notification somewhere else in this case when the socketFile is gone?

Reply:
These two do that, right?

```
				d.handleStaleWatchdogFiles()
				d.handleStaleSocketConnections()
```

Reply:
yup. so it's safe to ignore connection errors here. This resync is best effort, and the stale socket connections function handles the rest.